osclib/core.py

```diff
@@ -520,6 +526,14 @@ def project_meta_revision(apiurl, project):
         apiurl, project, '_project', None, format='xml', meta=True))
     return int(root.find('logentry').get('revision'))
 
+def package_source_changed(apiurl, project, package):
+    root = ETL.fromstringlist(get_commitlog(
```

Comment:
if you only need the latest history entry, calling get_commitlog is very expensive as it fetches the full history, parses it and creates custom XML out of it.

And _history has a limit parameter, but the osc function doesn't seem to support it. So I would instead fetch `/source/Kernel:stable/kernel-source/_history?limit=1` and take the time. This is what osc does to it:
```
        t = time.gmtime(int(node.find('time').text))
        t = time.strftime('%Y-%m-%d %H:%M:%S', t)
```

So just doing the time.gmtime will do, then you can also skip the date_parse

Reply:
Only called when actually updating a package, not on every package during check. Also just trying to use the `osc.core` API which is rather lacking as usual.

Reply:
It's a bit more fun to get the date objects to play nice without using date_parse() as the timezones are wrong, then one with and one without.